test/Polly.Core.Tests/Utils/Pipeline/ExecutionTrackingComponentTests.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Time.Testing;
+using Polly.Utils.Pipeline;
+
+namespace Polly.Core.Tests.Utils.Pipeline;
+
+public class ExecutionTrackingComponentTests
```

Comment:
Could we add a test that uses the strategy in a loop for a few seconds which gets triggered to dispose while it's running by a different thread as an extra validation on the current usage?

Reply:
Added tests that simulates multiple running tasks.